src/iracingdataapi/client.py

```diff
@@ -834,9 +834,9 @@ def result_search_series(
 
         params = locals()
         payload = {}
-        for x in params.keys():
-            if x != "self" and params[x]:
-                payload[x] = params[x]
+        for x, val in params.items():
+            if x != "self" and val is not None:
+                payload[x] = val
```

Comment:
This is good, thanks – if you can apply this to the other method too it'll be great.